DAOS-6222 dfs: set balanced mode with DTX as default

mchaarawi opened the pull request:
- change the default mode for DFS container create and mount to be balanced and use DTX

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
DAOS-6222 dfs: add container create flags for POSIX consistency

- Add relaxed & balanced (default) flags for dfs_cont_create
- update the daos tool to be able to set those flags on cont create
- add options to dfs_mount flags to set the mount flags
  - in this case, a container created with balanced can only be
    accessed with balanced mode.
  - a container created with relaxed mode can be accessed with
    either mode.
- add some simple unit tests to verify behavior.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 commented on src/client/dfs/dfs.c:
```diff
@@ -1022,7 +1033,8 @@ open_sb(daos_handle_t coh, bool create, dfs_attr_t *attr, daos_handle_t *oh)
 	}
 
-	if (iods[1].iod_size != sizeof(sb_ver) || sb_ver != DFS_SB_VERSION) {
+	/** check version compatiblity */
```
```suggestion
	/** check version compatibility */
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/1/execution/node/59/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daltonbohning commented on src/client/dfs/dfs.c:
```diff
@@ -1022,7 +1033,8 @@ open_sb(daos_handle_t coh, bool create, dfs_attr_t *attr, daos_handle_t *oh)
 	}
 
-	if (iods[1].iod_size != sizeof(sb_ver) || sb_ver != DFS_SB_VERSION) {
+	/** check version compatibility */
+	if (iods[1].iod_size != sizeof(sb_ver) || sb_ver > DFS_SB_VERSION) {
```
Minor/Prior Style (This wasn't introduced in this PR)
At first look, since `sb_ver` is set to 0 above, it seems that `sb_ver` will always be 0 in this check.
But further above, I see we're doing `d_iov_set(&sg_iovs[1], &sb_ver, sizeof(dfs_sb_ver_t));` and then after setting `sb_ver = 0` we make a call to `daos_obj_fetch`.
This seems a little confusing, though in hindsight I do understand what is going on.
Do we care about this? I.e. would it be better to initialize to 0 before
`d_iov_set(&sg_iovs[1], &sb_ver, sizeof(dfs_sb_ver_t));` to avoid confusion?

daltonbohning replied:
I would think not, unless maybe for some erroneous valgrind issue.

daltonbohning commented on src/client/dfs/dfs.c:
```diff
@@ -1123,6 +1138,11 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
 	else
 		dattr.da_oclass_id = DFS_DEFAULT_OBJ_CLASS;
 
+	if (attr && attr->da_mode == DFS_RELAXED)
+		dattr.da_mode = attr->da_mode;
+	else
+		dattr.da_mode = DFS_BALANCED;
+
```
Do we need to use `MODE_MASK` here just in case some other bits were set?

daltonbohning replied:
> maybe we need to fail in case it's set to some rubbish

I think that might be a good idea. I'm mostly considering what might happen if we _**do**_ happen to add some other flag  in the future and this line is overlooked.
Or if somehow we just accidentally reference/set the wrong variable at some point, and then this line overlooks that.

mchaarawi replied:
ok updated to check for that bit, and use balanced in default case.
the latter would still need some changes if we add another mode/flag anyway, but maybe this is clearer.

daltonbohning replied:
Yes, I think the update is much clearer. Thanks!

daltonbohning commented on src/tests/suite/dfs_unit_test.c:
```diff
@@ -84,6 +84,96 @@ dfs_test_mount(void **state)
+	rc = daos_cont_destroy(arg->pool.poh, cuuid, 1, NULL);
+	assert_int_equal(rc, 0);
+
+	/** create a DFS container with no mode specified */
```
Do we also want a test case here for "mount in Balanced / default mode should succeed", similar to when we explicitly create in balanced mode?

daltonbohning commented on src/utils/daos.c:
```diff
@@ -1317,6 +1341,9 @@ help_hdlr(int argc, char *argv[], struct cmd_args_s *ap)
 			"	--path=PATHSTR     container namespace path\n"
 			"container create common optional options:\n"
 			"	--type=CTYPESTR    container type (HDF5, POSIX)\n"
+			"	--mode=FLAG        In case of POSIX type, select consistency mode:\n"
+			"                          relaxed: weaker consistency semantics.\n"
+			"                          balanced (default): stronger consistency semantics.\n"
```
Minor: In my terminal, this displays as:
```
        --mode=FLAG        In case of POSIX type, select consistency mode:
                          relaxed: weaker consistency semantics.
                          balanced (default): stronger consistency semantics.
```
I think for the second and third lines, we also needs tabs instead of spaces. (at least for the part before --mode)

daltonbohning commented on src/utils/daos_hdlr.c:
```diff
@@ -1407,6 +1407,7 @@ cont_create_hdlr(struct cmd_args_s *ap)
 		attr.da_oclass_id = ap->oclass;
 		attr.da_chunk_size = ap->chunk_size;
 		attr.da_props = ap->props;
+		attr.da_mode = ap->mode;
```
Since the mode is only used for POSIX, do we want `daos.c` to print an error if a user gives a mode for a non-POSIX container?

daltonbohning replied:
Hmm. On second thought, it seems that `daos.c` doesn't check against this for other utilities in general. Maybe we don't care?

daltonbohning replied:
I see you updated this. Thanks!

daltonbohning reviewed (dismissed):
This seems to be working correctly; I just had some minor comments/questions.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
address review comments.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daltonbohning reviewed (dismissed):
LGTM.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/4/execution/node/705/log

ashleypittman commented:
I can review this tomorrow but the NLT errors are showing there's a problem with dfs_cont_create() not setting up the new container properly.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/5/execution/node/706/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
fix bug where attr struct is passed as NULL and values are not initialized

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Conflicts:
	src/utils/daos.c

mchaarawi committed:
merge conflict

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

ashleypittman commented on src/client/dfs/dfs.c:
```diff
@@ -1220,11 +1245,6 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 		D_GOTO(err_prop, rc = ENOMEM);
 
-	dfs->use_dtx = false;
-	d_getenv_bool("DFS_USE_DTX", &dfs->use_dtx);
```
There's a command line option in NLT which sets this env var, so there are three or four lines in that file that should be removed as well.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/10/execution/node/706/log

mchaarawi committed:
DAOS-6222 dfs: use TX in DFS open only when needed

- when using O_EXCL mode, a TX is not needed since it's only 1
  conditional op.
- Dir and Symlinks don't need a TX for the same reason.
- Refactor dfs_cont_create() a little and add a unit test for
  atomicity.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Conflicts:
	src/client/dfs/dfs.c
	src/tests/suite/dfs_unit_test.c

mchaarawi committed:
add debug logging for MPIO llnl test

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/12/execution/node/1114/log

daosbuild1 commented:
Test stage Functional on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/12/execution/node/1143/log

daosbuild1 commented:
Test stage Functional on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/13/execution/node/610/log

daosbuild1 commented:
Test stage Functional on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/13/execution/node/650/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Skip-nlt: true

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/14/execution/node/726/log

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/15/execution/node/316/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Revert "add debug logging for MPIO llnl test"

This reverts commit 01ecaf4669d12fd1ea8c0105beee359c942ed932.

Conflicts:
	src/tests/ftest/io/llnl_mpi4py.yaml

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
revert debugging logs for mdtest

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/19/execution/node/770/log

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/20/execution/node/291/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/21/execution/node/769/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Conflicts:
	src/client/dfs/dfs.c

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/execution/node/815/log

ashleypittman commented:
> Test stage NLT completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/execution/node/815/log

This is the same result NLT has been reporting for a bit on this PR, an attempt to clobber one file with another isn't working.
FileExistsError: [Errno 17] File exists: '/tmp/dfs_test/unlinked2' -> '/tmp/dfs_test/unlinked'

mchaarawi commented:
> > Test stage NLT completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/execution/node/815/log
> 
> This is the same result NLT has been reporting for a bit on this PR, an attempt to clobber one file with another isn't working.
> FileExistsError: [Errno 17] File exists: '/tmp/dfs_test/unlinked2' -> '/tmp/dfs_test/unlinked'

can you point me to the daos logs for this test?

ashleypittman commented:
> > > Test stage NLT completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/execution/node/815/log
> > 
> > 
> > This is the same result NLT has been reporting for a bit on this PR, an attempt to clobber one file with another isn't working.
> > FileExistsError: [Errno 17] File exists: '/tmp/dfs_test/unlinked2' -> '/tmp/dfs_test/unlinked'
> 
> can you point me to the daos logs for this test?

Immediately after the first assert in the output it logs "Running log_test on /tmp/dnt_dfuse_test_fchmod_replaced_q2dh6scs.log" which translates to https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/artifact/nlt_logs/dnt_dfuse_test_fchmod_replaced_q2dh6scs.log.bz2

The server logs are at https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-4013/22/artifact/nlt_logs/dnt_server_3bru799a.log

mchaarawi committed:
add a simple rename test to reproduce error in NLT

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/tests/suite/dfs_unit_test.c

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/24/execution/node/773/log

mchaarawi committed:
do not use conditional insert in the rename case when a DTX is used.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi commented:
actually please hold on reviews. i have an additional improvement i want to add here

mchaarawi committed:
Merge branch 'mschaara/dfs_open_tx' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/client/dfs/dfs.c

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 commented on src/client/dfs/dfs.c:
```diff
@@ -1273,20 +1329,18 @@
-	daos_obj_close(super_oh, NULL);
+	rc = daos_obj_close(super_oh, NULL);
+	if (rc) {
+		D_ERROR("Failed to clsoe SB object\n");
```
```suggestion
		D_ERROR("Failed to close SB object\n");
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/27/execution/node/69/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/28/execution/node/1409/log

mchaarawi committed:
in case of O_CREAT without O_EXCL, we should not use conditional insert
since it's already within a DTX.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
fix bug in open_file causing infinite loop

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/30/execution/node/815/log

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/31/execution/node/371/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
fix bug that does not commit on file creation without O_EXCL

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
fix bug on entry punch with dtx / conditionals

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/client/dfs/dfs.c
	src/tests/suite/dfs_unit_test.c
	src/utils/daos.c

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build on CentOS 7 debug completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/409/log

daosbuild1 commented:
Test stage Build on CentOS 7 release completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/388/log

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/320/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/335/log

daosbuild1 commented:
Test stage Build on Leap 15 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/411/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/321/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/38/execution/node/347/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/39/execution/node/787/log

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/40/execution/node/834/log

mchaarawi committed:
DAOS-623 object: Punch with valid dtx does not set the obj pointer

fix bug when dtx is used in object punch that leaves the obj NULL.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/42/execution/node/834/log

Nasf-Fan committed:
DAOS-7709 object: miss obj parameter when cache transactional list

That will cause object reference leak.

Signed-off-by: Fan Yong <fan.yong@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/client/dfs/dfs.c
	src/include/daos_fs.h

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/client/dfs/dfs.c
	src/include/daos_fs.h
	src/tests/suite/dfs_unit_test.c
	src/utils/daos.c

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
add reporducer for IOR -a POSIX test failure.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4013/46/execution/node/727/log

mchaarawi committed:
fix go test expected output for mode value

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
- change environment variable for DFS_USE_DTX to DFS_NO_DTX.
- update test

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4013/49/display/redirect

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

Conflicts:
	src/tests/suite/dfs_par_test.c

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4013/50/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

mchaarawi committed:
fix checkpatch

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

mchaarawi committed:
Merge branch 'master' into mschaara/dfs_modes

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

mchaarawi committed:
add test to emulate mdtest rename with branching factor

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
